auth/src/main/java/com/firebase/ui/auth/ui/account_link/SaveCredentialsActivity.java

```diff
@@ -60,11 +60,19 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.save_credentials_layout);
+
+        // If SmartLock is not enabled, finish immediately
+        if (!getIntent().getBooleanExtra(ExtraConstants.EXTRA_SMARTLOCK_ENABLED, true)) {
```

Comment:
Agreed.  Now that I'm looking at it there is no reason for SaveCredentialsActivity to be an Activity at all besides encapsulation.  It has no UI, I am going to refactor it.